### PR TITLE
Sample from active space for warmstart for random search

### PR DESCRIPTION
**Pull request by iaroslav-ai**
*Description of changes:*

In case of warmstart (active search space specified) candidates from active search space should be sampled for random search.

This also adds few instructions on how to develop this library.

### CODE REVIEW

File: README.md
Comment (iaroslav-ai) [inline README.md:43]: Without quotes install did not work for me. See also here https://stackoverflow.com/questions/52228286/how-to-pip-install-with-wheel-specifying-extras .

File: syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py
Comment (iaroslav-ai) [inline syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py:383]: Because sampling was done from "global" search space for warmstart, initial candidates would be out of bounds for active task.

File: tst/schedulers/bayesopt/test_hp_ranges.py
Comment (geoalgo) [inline tst/schedulers/bayesopt/test_hp_ranges.py:15]: Where do we we need this one?

File: README.md
Comment (geoalgo) [inline README.md:51]: Adding instructions for conda is a bit arbitrary, can we ommit it? (this raises the question why using conda and not venv, also we have to maintain those instructions)

File: README.md
Comment (geoalgo) [inline README.md:57]: It would be nice to use a more representative example, for instance `test_schedulers_api.py` that people will often need.

File: README.md
Comment (geoalgo) [reply README.md:43]: If we do this, we should also do it for all "pip install" instructions for uniformity.

File: syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py
Comment (mseeger) [reply syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py:383]: Thanks for catching this!

The best way to fix this is to expose the internal HyperparameterRanges._config_space_for_sampling as a property, and then use that. Do you want me to do this?

File: tst/schedulers/bayesopt/test_hp_ranges.py
Comment (iaroslav-ai) [reply tst/schedulers/bayesopt/test_hp_ranges.py:15]: My IDE added this without me noticing it :D Will remove.

File: README.md
Comment (mseeger) [reply README.md:51]: I am (very much) against conda, having had trouble with it in the past.

Comment (iaroslav-ai) [reply README.md:51]: Alright, will remove.

**Comment by iaroslav-ai** [comment]
Build is failing due to accidental `from distutils.command.config import config` - will push a fix in a moment.

File: README.md
Comment (iaroslav-ai) [reply README.md]: Used example from test_schedulers_api.py .

File: syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py
Comment (mseeger) [reply syne_tune/optimizer/schedulers/searchers/gp_fifo_searcher.py:383]: OK: #169 addresses this

**Comment by iaroslav-ai** [comment]
@mseeger  - once you merge your PR, I will update this to only contain changes to README, which I think are useful for someone new to repository.

**Review by mseeger — APPROVED** [review]
OK, let us take your PR, I'll close mine then.

But I am only approving to this change here (fix of sampling bug). For the others, please take David'scomments into account.

**Comment by iaroslav-ai** [comment]
Thank you Matthias! I will wait for David to comment / approve before merging.

File: README.md
Comment (iaroslav-ai) [reply README.md:43]: Added this to other places, and made sure instructions still work by running updated pip install commands.

**Pull request merged**
